test/Transport/TransportTest.re

```diff
@@ -30,17 +30,81 @@ let spawnNode = (~onExit, ~args) => {
   |> Result.get_ok;
 };
 
-let wait = condition => {
-  let start = Unix.gettimeofday();
-  let delta = () => Unix.gettimeofday() -. start;
+module Waiter = {
+  let wait = (~name="TODO", condition) => {
+    let start = Unix.gettimeofday();
+    let delta = () => Unix.gettimeofday() -. start;
+
+    while (!condition() && delta() < 1.0) {
+      let _: bool = Luv.Loop.run(~mode=`NOWAIT, ());
+      Unix.sleepf(0.1);
+    };
+
+    if (!condition()) {
+      failwith("Condition failed: " ++ name);
+    };
+  };
+
+  let waitForCollection = (~name, item) => {
+    let collected = ref(false);
+
+    let checkCollected = () => {
+      Gc.full_major();
+      collected^ == true;
+    };
+
+    Gc.finalise_last(() => collected := true, item);
+    wait(~name="Waiting for GC to collect: " ++ name, checkCollected);
+  };
+};
+
+module Test = {
+  type t = {
+    exits: ref(bool),
+    messages: ref(list(Transport.msg)),
+    transport: Transport.t,
+  };
+
+  let uniqueId = ref(0);
+
+  let start = scriptPath => {
+    let namedPipe =
+      NamedPipe.create("transport-test-" ++ (uniqueId^ |> string_of_int))
+      |> NamedPipe.toString;
+    incr(uniqueId);
+    let messages = ref([]);
+    let dispatch = msg => messages := [msg, ...messages^];
+
+    let exits = ref(false);
+    let onExit = (proc, ~exit_status, ~term_signal) => exits := true;
+    let _ = spawnNode(~onExit, ~args=[scriptPath, namedPipe]);
```

Comment:
Accidental partial application in waiting.

Reply:
Thanks for catching this! I wish I could set up the compiler to warn on this (ie, on `dune build @check`)